server/application/terminal.go

```diff
@@ -150,7 +153,8 @@ func (s *terminalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	pod, err := kubeClientset.CoreV1().Pods(namespace).Get(ctx, podName, metav1.GetOptions{})
 	if err != nil {
-		http.Error(w, "Cannot find pod: "+podName, http.StatusBadRequest)
+		fieldLog.Warn("Terminal Pod Not Found")
```

Comment:
This is an error and should be logged like so:
```Go
fieldLog.Errorf("error retrieving pod %s: %s", podName, err)
```

Reply:
@leoluz  - thank you for checking again - we are already logging the podName.  when `fieldLog` is called it will output ALL the fields (user, namespace, pod, container) to the log line.  since it is using `WithFields` - 
> Note that it doesn't log until you call Debug, Print, Info, Warn, Fatal or Panic on the Entry it returns.
I left as `"Terminal Pod Not Found"` to improve the searchability and to create standardized alerts based on that key - we can assume the format of that structured log will stay the same and build alerts based on that key.  

I defer to all the argocd folks but the distinction here which is important is user error vs application error.
It makes sense that this throws an http error to the user.
However it is not actually an application error - the application is performing normally.
From my perspective it would be better to be `Warn` instead of `Error` for application log level.
As an operator I would want to know that this is happening (`Warn`) but I would not consider this an indicator that argocd is unhealthy, having an issue, or doing something broken or unexpected.

Let me know if I should change it or if the above doesn't make sense.

Reply:
Agree with the fact that we don't need the pod name as it is already part of registered fields. However I believe this still needs to be logged as an error. It is ArgoCD internal code that does the call to kube-api to retrieve the pod by its name. If this request fails for some reason it is an internal error and should be logged like so. Retrieving the pod call can fail by several reasons and logging it as `terminal pod not found` is misleading.
I suggest to change this log to:
```Go
fieldLog.Errorf("error retrieving pod: %s", err)
```

By the way, log messages should be all lowercase sentences.

Reply:
@leoluz what if a Pod is in the resource tree but not synced? In that case an error from the k8s API would be a user error rather than an application error.

Reply:
@leoluz - good point - I have updated it and also changed the log messages to all lowercase

Reply:
> @leoluz what if a Pod is in the resource tree but not synced? In that case an error from the k8s API would be a user error rather than an application error.

My understanding is that (contrary to HTTP return codes) from the log perspective, it doesn't matter if the error was caused by the user or by something internal. Our code invoked a method that returned an error and in this case an error should be logged. However there is another "subtle" problem in this case. Client-go returns an error if for example the resource isn't found. This isn't an exceptional scenario and if we want to be precise to whether or not log an error (and decide the proper http return code) we need to do an extra check:
```Go
if err != nil {
    if apimachineryerrors.IsNotFound(err) {
        // don't log error
        // http 404
    } else {
        // log error
        // http 5xx
    }
}
```